src/main/java/com/force/api/ExtendedApiVersion.java

```diff
@@ -0,0 +1,95 @@
+package com.force.api;
+
+
+/**
+ * represents a single supported API version as reported by the <code>/services/data</code> resource
+ * for the instance endpoint returned on authentication.
+ *
+ * {@link com.frejo.api.ApiVersion} was originally used to set the current API version and manage the complete list
+ * of versions as an enum. It will likely be phased out in favor of using plain version strings along with this class
+ * for more extended version information.
+ *
+ * This is a tweak to an original proposal by @cswendrowski.
+ *
+ */
+public class ExtendedApiVersion implements Comparable<ExtendedApiVersion> {
+
+    public enum Season {
+        WINTER(),
+        SPRING(),
+        SUMMER();
+    }
+
+    public ExtendedApiVersion() {}
+
+    public ExtendedApiVersion(float version) {
+        this.version = version;
+    }
+
+    public ExtendedApiVersion(Season s, int y) {
```

Comment:
These variables should be expanded out to `season` and `year` to match the rest of the code style.

Fun side note - some IDE's such as IntelliJ will now show the variable name in-line for Primitives when calling methods for easier understanding of logic flow:

![http://puu.sh/ugphA/a20c173e08.png](http://puu.sh/ugphA/a20c173e08.png)